fiasco/ions.py

```diff
@@ -1576,48 +1576,40 @@ def two_photon(self, wavelength: u.angstrom,
         include_protons : `bool`, optional
             If True, use proton excitation and de-excitation rates in the level population calculation.
         """
-
         wavelength = np.atleast_1d(wavelength)
-        temperature = np.atleast_1d(self.temperature)
         electron_density = np.atleast_1d(electron_density)
 
-        prefactor = (const.h * const.c)
-
-        if not self.hydrogenic and not self.helium_like:
-            return u.Quantity(np.zeros(wavelength.shape + self.temperature.shape + electron_density.shape),
-                              'erg cm^3 s^-1 Angstrom^-1')
+        final_shape = wavelength.shape + self.temperature.shape + electron_density.shape
         if self.hydrogenic:
             A_ji = self._hseq['A']
             psi_norm = self._hseq['psi_norm']
             cubic_spline = CubicSpline(self._hseq['y'], self._hseq['psi'])
-            # Get the index of the 2S1/2 state for H-like:
-            level_index = np.where((self._elvlc['config'] == '2s') & (self._elvlc['J'] == 0.5))[0][0]
-        if self.helium_like:
+            config = '2s'  # Get the index of the 2S1/2 state for H-like
+            J = 0.5
```

Comment:
Now that I'm thinking about it, could there be an issue with floating precision with `J`?  

That is, there's a statement `self._elvlc['J'] == 0.5`, which could perhaps be `math.isclose(self._elvlc['J'], 0.5)` ?

Reply:
Well, `math.isclose(self._elvlc['J'], J)` in the new changes.

Reply:
I updated this one myself (to use `np.allclose`) as I'm certain we should not use `==` for float comparisons.

Reply:
Fair point. This makes me think we should representing these spin/angular momentum numbers as `Fraction` objects (https://docs.python.org/3/library/fractions.html) or something similar, but that is for another PR.